libs/sr-ui/src/Character/CharacterAutocomplete.tsx

```diff
@@ -16,49 +15,34 @@ export function CharacterAutocomplete({
   charKey,
   setCharKey,
 }: CharacterAutocompleteProps) {
+  const { t } = useTranslation(['character', 'charNames_gen'])
   const { database } = useDatabaseContext()
-  const [charListDirty, setCharListDirty] = useForceUpdate()
-  useEffect(
-    () => database.chars.followAny(() => setCharListDirty()),
-    [database, setCharListDirty]
-  )
-  const [charFaveDirty, setCharFaveDirty] = useForceUpdate()
-  useEffect(
-    () => database.charMeta.followAny(() => setCharFaveDirty()),
-    [database, setCharFaveDirty]
-  )
-
-  const charDbMap = useMemo(
-    () => charListDirty && objKeyMap(database.chars.keys, () => true),
-    [charListDirty, database.chars.keys]
+  const charInDb = useCallback(
```

Comment:
This is a good example of "code optimization before there is any need". I am sure in this case the useMemo is actually adding onto both code complexity and runtime performance. Good catch. We should make an effort to see if this can be amended in some of the other autocompletes as well.